### ISSUE TITLE
Don't pass empty handler to Java function

### ISSUE THREAD

**Issue by trisberg** [projectriff/riff-buildpack#18]
The buildpack should not set the `handler` argument if handler is empty. This causes an NPE:

```
Caused by: java.lang.NullPointerException
	at org.springframework.cloud.function.deployer.FunctionCreatorConfiguration$BeanCreator.getFunctionNames(FunctionCreatorConfiguration.java:392)
	at org.springframework.cloud.function.deployer.FunctionCreatorConfiguration.functionNames(FunctionCreatorConfiguration.java:138)
	at org.springframework.cloud.function.deployer.FunctionCreatorConfiguration.init(FunctionCreatorConfiguration.java:119)
	... 24 more
```

Instead of passing in `--function.uri='file:///workspace/app?handler='` it would be better to pass in `--function.uri='file:///workspace/app'`

**Comment by ericbottard** [comment]
That would be equivalent on the java side (or any language for that matter) URL-wise wouldn't it?

**Comment by trisberg** [comment]
Not sure what you are asking , the `function.uri` is specific to Java, I'll add a fix to my PR to upgrade to Java invoker 0.1.3 since that version supports leaving out the handler if there is a single function bean

**Comment by ericbottard** [comment]
What I'm saying is that from the consuming side, empty value or no key at all have the same semantics (at least in URL/URI). It doesn't hurt to take care of this at both the producing and consuming side, but we shouldn't have to. The java invoker only needs to protect against it

**Comment by ericbottard** [comment]
Addressed by 28f94e1
